docker/config_template.yaml

```diff
@@ -20,6 +20,7 @@ persistence:
                 keyspace: "{{ default .Env.KEYSPACE "temporal" }}"
                 user: "{{ default .Env.CASSANDRA_USER "" }}"
                 password: "{{ default .Env.CASSANDRA_PASSWORD "" }}"
+                allowedAuthenticators: "{{ default .Env.CASSANDRA_ALLOWED_AUTHENTICATORS "" }}"
```

Comment:
This won't unmarshal properly, the yaml needs to provide a list here. I tested by adding this to `config/development.yaml`:

```
  datastores:
    cassandra:
      cassandra:
        allowedAuthenticators: "one, two"
```

and then running the `config/config_test.go`

```--- FAIL: TestToString (0.00s)
    /Users/robholland/Developer/github.com/temporalio/temporal/common/config/config_test.go:37: 
        	Error Trace:	/Users/robholland/Developer/github.com/temporalio/temporal/common/config/config_test.go:37
        	Error:      	Received unexpected error:
        	            	yaml: unmarshal errors:
        	            	  line 12: cannot unmarshal !!str `one, two` into []string
        	Test:       	TestToString
```

You will need to use use a range and dockerize's `split` helper here to provide an array with a value per authenticator.

Reply:
Good catch. Fixed it. Here is how I tested it.
1. `export CASSANDRA_ALLOWED_AUTHENTICATORS="one, two"`
2. `dockerize -template ./docker/config_template.yaml:/tmp/qqq.yaml && cat /tmp/qqq.yaml` - Verify that the allowedAuthenticators field is populated as a list.
3. Update `config/development.yaml` with the values generated above and run - `go test ./common/config`

Reply:
I think we should have config tests for cassandra as well in our unit tests. It's outside the scope of this PR and needs a bit of test setup. I'll add it separately.

Reply:
I agree about the need and being out of scope.